src/gt4py/next/iterator/embedded.py

```diff
@@ -1496,6 +1496,48 @@ def _validate_domain(domain: Domain, offset_provider: OffsetProvider) -> None:
             )
 
 
+@runtime.set_at.register(EMBEDDED)
+def set_at(expr, domain, target) -> None:
+    # domain is assumed to match the domain of outer apply stencil / fields
+    if callable(expr):  # lazy apply stencil #TODO check
+        expr(target, lazy_domain=domain)
+
+
+@builtins.as_fieldop.register(EMBEDDED)
+def as_fieldop(fun, domain=None):
+    def impl(*args, **kwargs):
+        new_domain = None
+        # if common.is_domain_like(domain):
+        new_domain = domain
+        # else:
+        #     assert callable(domain)
+        #     new_domain = domain(*args, **kwargs)
+
+        # TODO this only works if the as_fieldop is directly in set_at
+        # for the clean solution we need to pre-allocate the result buffer (see strategy for scan in field_view embedded)
```

Comment:
We could use type annotation later (would this work today for scans in fieldview embedded?)